docs/modules/ROOT/pages/index.adoc

```diff
@@ -0,0 +1,175 @@
+
+=== Requirements
+
+==== GMP for fastecdsa
+
+Before installing Cairo on your machine, you need to install `gmp`:
+
+[,bash]
+----
+sudo apt install -y libgmp3-dev # linux
+brew install gmp # mac
+----
+
+TIP: If you have any trouble installing gmp on your Apple M1 computer, https://github.com/OpenZeppelin/nile/issues/22[here's a list of potential solutions].
+
+==== Supported Python versions
+
+Some Nile dependencies have specific python version requirements. Therefore we recommend using a python version manager like {pyenv}, and virtual environments to avoid conflicts.
+
+IMPORTANT: Currently supported Python versions are >=3.8 and <3.10.
+
+=== Installation
+
+Create a folder for your project and cd into it:
+
+[,sh]
+----
+mkdir myproject && cd myproject
+----
+
+Create a virtualenv and activate it:
+
+[,sh]
+----
+python3 -m venv env
+source env/bin/activate
+----
+
+IMPORTANT: Review xref:index.adoc#requirements[Requirements]
+
+Install Nile:
+
+[,sh]
+----
+(env): pip install cairo-nile
+----
+
+Use `nile init` to quickly set up your development environment:
+
+[,sh]
+----
+(env): nile init
+🗄 Creating project directory tree
+⛵️ Nile project ready! Try running:
+
+nile compile
+----
+
+=== Testing
+
+`nile init` creates a sample Cairo contract and test for you. Check *contracts/contract.cairo* and *tests/test_contract.py* for the source code.
+
+Run `pytest` to run the test suite against the Smart Contracts:
+
+[,sh]
+----
+(env): pytest tests/
+----
+
+TIP: For a more in-deep guide on testing with parallelism and coverage, check our xref:testing.adoc[Testing] guide.
+
+=== Compiling
+
+Use `nile compile` to compile contracts under the *contracts/* folder by default.
+
+[,sh]
+----
+(env): nile compile
+🤖 Compiling all Cairo contracts in the contracts directory
+🔨 Compiling contracts/contract.cairo
+✅ Done
+----
+
+TIP: For a full reference of Nile command options, check the xref:commands.adoc[CLI Reference] section.
+
+=== Deploying an account
+
+. Add an environment variable with the Account Private Key to the *.env* file.
++
+[,sh]
+----
+# .env
+ACCOUNT_1 = 0x1234
+----
++
+CAUTION: Don't use this Private Key in production environments!
++
+. Use `nile counterfactual-address` to precompute the deployment address of the Account contract for the given signer.
++
+[,sh]
+----
+(env): nile counterfactual-address ACCOUNT_1 --salt 123
+0x057792c44b1e9e349ee549ed018f7225f927bd2b01b6277a1cb94da5fb9421ed
+----
++
+NOTE: Different salts compute to different addresses, even with the same Private Key.
++
+. In production, you will need to fund this address with Ether to pay for the deployment fees, which will be mandatory in testnets soon. There are multiple bridges and faucets, like link:https://goerli.starkgate.starknet.io/[Starkgate] being implemented at the moment to support the developer experience.
```

Comment:
non production users need instructions too, or they might get stuck/drop in here. it's a good place to link the "Transfer funds from predeployed devnet account" sample script

Reply:
Added an entry for this.